### PR TITLE
resource/apigatewayv2: Add support for `body` parameter

### PR DESCRIPTION
**Pull request by vincenting**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #11148 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/apigatewayv2: Add support for `body` parameter
```

Output from acceptance testing:

```
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (86.59s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (91.46s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	93.017s

...
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
@vincenting Nice solution to the problem 👏.

Validated acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (46.91s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (48.63s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	48.675s
```

It looks like you should be able to refactor the `ReimportApi`/`UpdateApi` blocks into a single function that is called from both Create and Update and can use `d.Get("")` instead of having to do a  `GetApi` call or use fields from the `CreateApi` response.

**Comment by vincenting** [comment]
@ewbankkit Thanks for your reply. 

I will try to use `d.Get('')` method and refactor a new function for that today.

**Comment by vincenting** [comment]
The acceptance testing output of the refactored version:

```
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (128.79s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (136.20s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	137.708s
```

**Comment by vincenting** [comment]
The testing output for the refactored version:

```
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (125.21s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (133.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	134.820s
```

File: aws/resource_aws_apigatewayv2_api.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ func resourceAwsApiGatewayV2Api() *schema.Resource {
 				Required:     true,
 				ValidateFunc: validation.StringLenBetween(1, 128),
 			},
+			"body": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (ewbankkit) [inline aws/resource_aws_apigatewayv2_api.go]: Can add
```go
				ValidateFunc:     validation.ValidateJsonString,
				DiffSuppressFunc: suppressEquivalentJsonDiffs,
```

Comment (vincenting) [reply aws/resource_aws_apigatewayv2_api.go]: Hi, as it documents, [You can write OpenAPI definitions in YAML or JSON](https://swagger.io/docs/specification/basic-structure/).

In order to improve the validate and different method here, we might need to implement a new pair of functions for OpenAPI only, which can also be used in `aws/resource_aws_api_gateway_rest_api.go`.

In that case, that field should meet either `StringIsJSON` or `StringIsYAML`. And the same for the  `DiffSuppressFunc `.

Comment (vincenting) [reply aws/resource_aws_apigatewayv2_api.go]: We might need another merge request to implement those validate and diff functions first for the  `aws/resource_aws_api_gateway_rest_api.go` then reuse it here.

I will try that tomorrow. 😃

**Comment by vincenting** [comment]
```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (83.98s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (91.44s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       93.078s
```

Some updates since last commit:
1. Reuse the diff and validate function for  CloudFormationTemplateBody. 
2. Change the tests and add YAML format OAI.

File: aws/diff_suppress_funcs.go [hunk 2]
```diff
@@ -125,6 +125,8 @@ func suppressCloudFormationTemplateBodyDiffs(k, old, new string, d *schema.Resou
 	return normalizedOld == normalizedNew
 }
 
+var suppressOpenAPISpecificationDiffs = suppressCloudFormationTemplateBodyDiffs
```
Comment (ewbankkit) [inline aws/diff_suppress_funcs.go:128]: Probably best for long term maintainability is to rename `suppressCloudFormationTemplateBodyDiffs` as `suppressEquivalentJsonOrYamlDiffs` and use that.

File: aws/validators.go [hunk 3]
```diff
@@ -897,6 +897,8 @@ func validateCloudFormationTemplate(v interface{}, k string) (ws []string, error
 	return
 }
 
+var validateOpenAPISpecification = validateCloudFormationTemplate
```
Comment (ewbankkit) [inline aws/validators.go:900]: As for `diff_suppress_funcs.go`, I suggest renaming `validateCloudFormationTemplate` to `validateJsonOrYaml`.

**Comment by DrFaust92** [comment]
It seems that https://github.com/terraform-providers/terraform-provider-aws/pull/12715 may introduce some changes

**Comment by ewbankkit** [comment]
@vincenting Now that all the other APIGW v2/HTTP API PRs have been merged, do you have time to rebase and fix the merge conflicts? Many thanks.

**Comment by vincenting** [comment]
@ewbankkit Sure. I will work on that soon.

**Comment by vincenting** [comment]
```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (147.15s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (161.70s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       163.500s
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$  make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (47.04s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (48.88s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	48.929s
```

**Comment by ewbankkit** [comment]
@gdavison Can we line this one up for review please?

**Comment by hp0210** [comment]
Please review and merge this as soon as you guys can. We really need this to simplify our API configuration

**Comment by ewbankkit** [comment]
@vincenting Sorry for the delay here.
Do you have time to rebase and fix the merge conflict?
Thanks.

File: website/docs/r/apigatewayv2_api.html.markdown [hunk 4]
```diff
@@ -52,8 +52,11 @@ Defaults to `$request.method $request.path`.
 * `target` - (Optional) Part of _quick create_. Quick create produces an API with an integration, a default catch-all route, and a default stage which is configured to automatically deploy changes.
 For HTTP integrations, specify a fully qualified URL. For Lambda integrations, specify a function ARN.
 The type of the integration will be `HTTP_PROXY` or `AWS_PROXY`, respectively. Applicable for HTTP APIs.
+* `body` - (Optional) An OpenAPI specification that defines the set of routes and integrations to create as part of the HTTP APIs. Supported only for HTTP APIs.
 * `version` - (Optional) A version identifier for the API.
 
+__Note__: If the `body` argument is provided, the `info` field will also be used to update the attributes of the API Gateway by AWS. It's a good practice to keep them with the same values. However, to avoid further issues, We will not respect those definitions in your `info` field intentionally.
```
Comment (ewbankkit) [inline website/docs/r/apigatewayv2_api.html.markdown:58]: Could you clarify what you mean by `info` field here?

Also, could you add a note similar to the REST API equivalent

```
__Note__: If the `body` argument is provided, the OpenAPI specification will be used to configure the resources, methods and integrations for the Rest API. If this argument is provided, the following resources should not be managed as separate ones, as updates may cause manual resource updates to be overwritten:

* `aws_api_gateway_resource`
* `aws_api_gateway_method`
* `aws_api_gateway_method_response`
* `aws_api_gateway_method_settings`
* `aws_api_gateway_integration`
* `aws_api_gateway_integration_response`
* `aws_api_gateway_gateway_response`
* `aws_api_gateway_model`
```

about which other resources are implicitly created and that they _shouldn't_ be `terraform import`ed as they are managed via the API spec?
I think at least `aws_apigatewayv2_route` and `aws_apigatewayv2_integration` get implicitly created.

Comment (vincenting) [reply website/docs/r/apigatewayv2_api.html.markdown:58]: Sure, I will add it this weekend.

File: aws/resource_aws_apigatewayv2_api.go [hunk 5]
```diff
@@ -135,6 +141,39 @@ func resourceAwsApiGatewayV2Api() *schema.Resource {
 	}
 }
 
+func resourceAwsAPIGatewayV2ImportOpenAPI(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigatewayv2conn
+
+	if body, ok := d.GetOk("body"); ok {
+		revertReq := &apigatewayv2.UpdateApiInput{
```
Comment (ewbankkit) [inline aws/resource_aws_apigatewayv2_api.go:148]: Can you please note in the documentation that `name`, `description` and `version` should be specified in the Terraform configuration and the values will override any values specified in the OpenAPI document.
`cors_configuration` - https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-cors-configuration.html - and `tags` - https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-openapi-extensions-x-amazon-apigateway-tag-value.html - can also be specified in the OpenAPI document so we should have the same behaviour for those attributes.

File: aws/resource_aws_apigatewayv2_api.go [hunk 6]
```diff
@@ -135,6 +141,39 @@ func resourceAwsApiGatewayV2Api() *schema.Resource {
...
+		log.Printf("[DEBUG] Updating API Gateway from OpenAPI spec %s", d.Id())
+		importReq := &apigatewayv2.ReimportApiInput{
+			ApiId: aws.String(d.Id()),
+			Body:  aws.String(body.(string)),
+		}
+
+		_, err := conn.ReimportApi(importReq)
+
+		if err != nil {
+			return fmt.Errorf("error importing API Gateway specification: %s", err)
```
Comment (ewbankkit) [inline aws/resource_aws_apigatewayv2_api.go:164]:
```suggestion
			return fmt.Errorf("error importing API Gateway v2 API (%s) OpenAPI specification: %s", d,Id(), err)
```

File: aws/resource_aws_apigatewayv2_api.go [hunk 7]
```diff
@@ -135,6 +141,39 @@ func resourceAwsApiGatewayV2Api() *schema.Resource {
...
+		}
+
+		log.Printf("[DEBUG] Reverting API Gateway v2 API: %s", revertReq)
+		_, err = conn.UpdateApi(revertReq)
+		if err != nil {
+			return fmt.Errorf("error reverting API Gateway v2 API (%s): %s", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_apigatewayv2_api.go:170]:
```suggestion
			return fmt.Errorf("error updating API Gateway v2 API (%s): %s", d.Id(), err)
```

It's good to have `reverting` in the log statement but would be confusing in the error message.

File: aws/resource_aws_apigatewayv2_api.go [hunk 8]
```diff
@@ -177,6 +216,11 @@ func resourceAwsApiGatewayV2ApiCreate(d *schema.ResourceData, meta interface{})
 
 	d.SetId(aws.StringValue(resp.ApiId))
 
+	err = resourceAwsAPIGatewayV2ImportOpenAPI(d, meta)
+	if err != nil {
+		return fmt.Errorf("error update API Gateway v2 API (%s) with OpenAPI spec: %s", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_apigatewayv2_api.go:221]:
```suggestion
		return err
```

The relevant error information was captured in `resourceAwsAPIGatewayV2ImportOpenAPI` so there's no need to wrap here.

File: aws/resource_aws_apigatewayv2_api.go [hunk 9]
```diff
@@ -278,6 +322,13 @@ func resourceAwsApiGatewayV2ApiUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	if d.HasChange("body") {
+		err := resourceAwsAPIGatewayV2ImportOpenAPI(d, meta)
+		if err != nil {
+			return fmt.Errorf("error update API Gateway v2 API (%s) with OpenAPI spec: %s", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_apigatewayv2_api.go:328]:
```suggestion
			return err
```

The relevant error information was captured in `resourceAwsAPIGatewayV2ImportOpenAPI` so there's no need to wrap here.

**Comment by ewbankkit** [comment]
@vincenting This is very close to being ready to merge, Do you have time to make the minor review changes and documentation update? It would be great to get these into this week's release and close out #11148.

**Comment by vincenting** [comment]
> @vincenting This is very close to being ready to merge, Do you have time to make the minor review changes and documentation update? It would be great to get these into this week's release and close out #11148.

Hi, sorry for being a bit busy recently. I will try to fix those tomorrow to catch up with the release.

**Comment by vincenting** [comment]
```
$make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Api_Openapi'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_Openapi -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithTags
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (88.34s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithTags (90.14s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (96.80s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration (146.08s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	147.866s
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Api_disappears
=== PAUSE TestAccAWSAPIGatewayV2Api_disappears
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== RUN   TestAccAWSAPIGatewayV2Api_Tags
=== PAUSE TestAccAWSAPIGatewayV2Api_Tags
=== RUN   TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_QuickCreate
=== PAUSE TestAccAWSAPIGatewayV2Api_QuickCreate
=== CONT  TestAccAWSAPIGatewayV2Api_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_disappears
=== CONT  TestAccAWSAPIGatewayV2Api_basicHttp
=== CONT  TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_QuickCreate
=== CONT  TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
--- PASS: TestAccAWSAPIGatewayV2Api_basicWebSocket (13.91s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicHttp (14.11s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (25.47s)
--- PASS: TestAccAWSAPIGatewayV2Api_QuickCreate (26.40s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration (51.90s)
--- PASS: TestAccAWSAPIGatewayV2Api_disappears (55.55s)
--- PASS: TestAccAWSAPIGatewayV2Api_CorsConfiguration (58.26s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithTags (60.69s)
--- PASS: TestAccAWSAPIGatewayV2Api_Tags (63.67s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesHttp (64.55s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket (64.73s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (82.43s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	82.538s
```

File: aws/validators_test.go [hunk 10]
```diff
@@ -2,11 +2,12 @@ package aws
...
 import (
 	"fmt"
-	"github.com/aws/aws-sdk-go/service/cognitoidentity"
-	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
 	"regexp"
 	"strings"
 	"testing"
+
+	"github.com/aws/aws-sdk-go/service/cognitoidentity"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
```
Comment (bflad) [inline aws/validators_test.go:10]: Hi @vincenting 👋  Can you please rebase this pull request? I'm guessing our recent removal of SDKv1 is causing the merge conflict here (the new path is `github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest` -- apologies it affected this contribution.

**Comment by vincenting** [comment]
```bash
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Api_disappears
=== PAUSE TestAccAWSAPIGatewayV2Api_disappears
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== RUN   TestAccAWSAPIGatewayV2Api_Tags
=== PAUSE TestAccAWSAPIGatewayV2Api_Tags
=== RUN   TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_QuickCreate
=== PAUSE TestAccAWSAPIGatewayV2Api_QuickCreate
=== CONT  TestAccAWSAPIGatewayV2Api_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== CONT  TestAccAWSAPIGatewayV2Api_disappears
=== CONT  TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_QuickCreate
=== CONT  TestAccAWSAPIGatewayV2Api_basicHttp
2020/08/14 08:25:19 [DEBUG] Trying to get account information via sts:GetCallerIdentity
    TestAccAWSAPIGatewayV2Api_disappears: resource_aws_apigatewayv2_api_test.go:149: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSAPIGatewayV2Api_disappears (85.94s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicWebSocket (94.64s)
--- PASS: TestAccAWSAPIGatewayV2Api_QuickCreate (112.36s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicHttp (121.16s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (139.12s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithTags (146.88s)
--- PASS: TestAccAWSAPIGatewayV2Api_Tags (152.66s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (161.56s)
--- PASS: TestAccAWSAPIGatewayV2Api_CorsConfiguration (167.05s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration (173.35s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket (195.04s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesHttp (205.04s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	206.673s
```

**Comment by ewbankkit** [comment]
Verified acceptance tests after rebase:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Api_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Api_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Api_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Api_disappears
=== PAUSE TestAccAWSAPIGatewayV2Api_disappears
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== RUN   TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== PAUSE TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== RUN   TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== PAUSE TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== RUN   TestAccAWSAPIGatewayV2Api_Tags
=== PAUSE TestAccAWSAPIGatewayV2Api_Tags
=== RUN   TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== PAUSE TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== RUN   TestAccAWSAPIGatewayV2Api_QuickCreate
=== PAUSE TestAccAWSAPIGatewayV2Api_QuickCreate
=== CONT  TestAccAWSAPIGatewayV2Api_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_QuickCreate
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi_WithTags
=== CONT  TestAccAWSAPIGatewayV2Api_Openapi
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesHttp
=== CONT  TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket
=== CONT  TestAccAWSAPIGatewayV2Api_disappears
=== CONT  TestAccAWSAPIGatewayV2Api_CorsConfiguration
=== CONT  TestAccAWSAPIGatewayV2Api_Tags
=== CONT  TestAccAWSAPIGatewayV2Api_basicHttp
=== CONT  TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields
=== CONT  TestAccAWSAPIGatewayV2Api_disappears
    resource_aws_apigatewayv2_api_test.go:149: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSAPIGatewayV2Api_disappears (28.65s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicHttp (37.95s)
--- PASS: TestAccAWSAPIGatewayV2Api_QuickCreate (40.57s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithTags (74.00s)
--- PASS: TestAccAWSAPIGatewayV2Api_basicWebSocket (77.68s)
--- PASS: TestAccAWSAPIGatewayV2Api_CorsConfiguration (85.67s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi_WithCorsConfiguration (98.87s)
--- PASS: TestAccAWSAPIGatewayV2Api_Tags (99.38s)
--- PASS: TestAccAWSAPIGatewayV2Api_Openapi (104.89s)
--- PASS: TestAccAWSAPIGatewayV2Api_OpenapiWithMoreFields (107.51s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesWebSocket (115.26s)
--- PASS: TestAccAWSAPIGatewayV2Api_AllAttributesHttp (122.67s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	122.723s
```

LGTM.

**Pull request merged**